// ngx-page-scroll-core/esm2020/lib/ngx-page-scroll-core.module.mjs
import { NgModule } from '@angular/core';
import { PageScrollService } from './providers/ngx-page-scroll.service';
import { NGXPS_CONFIG } from './providers/config.provider';
import * as i0 from "@angular/core";
export class NgxPageScrollCoreModule {
    static forRoot(config) {
        return {
            ngModule: NgxPageScrollCoreModule,
            providers: [PageScrollService, { provide: NGXPS_CONFIG, useValue: config }],
        };
    }
}
NgxPageScrollCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.2", ngImport: i0, type: NgxPageScrollCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxPageScrollCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.2", ngImport: i0, type: NgxPageScrollCoreModule });
NgxPageScrollCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.2", ngImport: i0, type: NgxPageScrollCoreModule, providers: [
        PageScrollService,
        { provide: NGXPS_CONFIG, useValue: {} },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.2", ngImport: i0, type: NgxPageScrollCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        PageScrollService,
                        { provide: NGXPS_CONFIG, useValue: {} },
                    ],
                }]
        }] });
